pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/PdfDao.java

```diff
@@ -268,13 +269,13 @@ private void deleteAnswerSubstitution(AnswerSubstitution substitution) {
      */
     default boolean deleteSpecificConfigVersion(PdfConfiguration config) {
         PdfVersion version = config.getVersion();
-        List<PdfTemplate> templates = config.getTemplates();
+        List<Long> templateIds = config.getTemplateIds();
 
         PdfSql pdfSql = getPdfSql();
-        DBUtils.checkDelete(templates.size(), pdfSql.unassignTemplatesFromVersion(version.getId()));
+        DBUtils.checkDelete(templateIds.size(), pdfSql.unassignTemplatesFromVersion(version.getId()));
 
-        for (PdfTemplate template : templates) {
-            deleteTemplate(template);
+        for (Long templateId : templateIds) {
+            findFullTemplateByTemplateId(templateId).ifPresent(template -> deleteTemplate(template));
```

Comment:
Might be better to do `orElseThrow()` and calling `deleteTemplate()` separately.

Reply:
Sure 0892b1c0